frontend/src/routes/GymPage/index.tsx

```diff
@@ -49,22 +57,77 @@ export default function GymPage({ onLeftClick }: { onLeftClick?: () => void }) {
       userID: 'A0234567M',
     },
   ]
+  useEffect(() => {
+    dispatch(getGymStatus())
+    dispatch(getUserDetail())
+    // dispatch(getGymHistory())
+  }, [])
 
   const [currentTab, setCurrentTab] = useState<number>(1)
   const sections = ['Gym', 'History']
   const history = useHistory()
+  //TODO: export this
+  const buttonText = ['Key With Me', 'Return Key', 'Close Gym', 'Open Gym']
+
+  function getButtonStates(): ButtonStates {
+    if (!gymStatus.gymIsOpen) {
+      if (gymStatus.keyHolder.displayName == name) {
+        return { keyWithMe: false, returnKey: true, toggleGym: true }
```

Comment:
consider abstracting them out into a variable for easy reading.

const gymOpenBySomeoneElseButtonStates = {..false ..true ..true}
const gymOpenByMeButtonStates = {......}
const gymClosedButtonStates = {...}

if (xxx) {
  return gymClosedButtonStates